documentation/modules/exploit/linux/http/artica_proxy_unauth_rce_cve_2024_2054.md

```diff
@@ -0,0 +1,213 @@
+## Vulnerable Application
+
+A Command Injection vulnerability in Artica Proxy appliance version `4.50` and `4.40`
+allows remote attackers to run arbitrary commands via unauthenticated HTTP request.
+The Artica Proxy administrative web application will de-serialize arbitrary PHP objects
+supplied by unauthenticated users and subsequently enable code execution as the `www-data` user.
+
+This module has been tested with:
+* Artica Proxy Appliance 4.50 running in VirtualBox 7.0.14 r161095 (Qt5.15.2)
+* Artica Proxy Appliance 4.40 Service Pack 118 running in VirtualBox 7.0.14 r161095 (Qt5.15.2)
+
+## Installation steps to install Artica Proxy appliance
+* Install your favorite virtualization engine (VMware or VirtualBox) on your preferred platform.
+* Here are the installation instructions for [VirtualBox on MacOS](https://tecadmin.net/how-to-install-virtualbox-on-macos/).
+* Download the Artica Proxy iso image from [here](https://sourceforge.net/projects/artica-squid/files/ISO/).
+* Install the iso image in your virtualization engine.
+* When installed, configure the VM appliance to your needs using the menu options.
+* Boot up the VM and should be able to access the Artica  appliance.
+* Either thru the console, `ssh` on port `22` or via the `webui` via `https://your_articaproxy_ip:9000`.
+
+You are now ready to test the module.
+
+## Verification Steps
+
+- [ ] Start `msfconsole`
+- [ ] `use exploit/linux/http//artica_proxy_unauth_rce_cve_2024_2054`
+- [ ] `set rhosts <ip-target>`
+- [ ] `set rport <port>`
+- [ ] `set webshell cuckoo`
+- [ ] `set target <0=PHP, 1=Unix Command, 2=Linux Dropper>`
+- [ ] `exploit`
+- [ ] you should get a `reverse shell` or `Meterpreter` session depending on the `payload` and `target` settings
+
+## Options
+
+### WEBSHELL
+You can use this option to set the filename without extension of the webshell.
+This is handy if you want to test the webshell upload and execution with different file names.
+to bypass any security settings on the Web and PHP server.
+
+### COMMAND
+This option provides the user to choose the PHP underlying shell command function to be used for execution.
+The choices are `system()`, `passthru()`, `shell_exec()` and `exec()` and it defaults to `passthru()`.
+This option is only available when the target selected is either Unix Command or Linux Dropper.
+For the native PHP target, by default the `eval()` function will be used for native PHP code execution.
+
+## Scenarios
+
+```shell
```

Comment:
Using `msf` here will allow it be rendered as msfconsole output on our docs site.

```suggestion
```msf
```

Reply:
See https://github.com/rapid7/metasploit-framework/pull/18967/commits/f217312ad1beab521059c172cd23fb8caa661618.